.github/workflows/upgrade-tests-reusable.yaml

```diff
@@ -0,0 +1,106 @@
+name: Upgrade Tests Reusable Workflow
+
+env:
+  E2E_LOG_LEVEL: debug
+  KYMA_STABILITY: "unstable"
+  KYMA: "./hack/kyma"
+
+on:
+  workflow_call:
+    inputs:
+      pre-upgrade-image-tag:
+        required: true
+        type: string
+      post-upgrade-image-tag:
+        required: true
+        type: string
+      build-job-name:
+        type: string
+      build-job-commit-sha:
+        type: string
+
+jobs:
+  e2e-upgrade: # This job tests the upgrade of Eventing module from latest image of the main branch to the current commit.
+    runs-on: ubuntu-latest
+    env:
+      BACKEND_TYPE: NATS
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Pre-upgrade checkout
+        run: git checkout -b ${{ inputs.pre-upgrade-image-tag }}
```

Comment:
why not use `ref` for `actions/checkout@v4`

Reply:
You are right, it works in this case. Just wanted to make consistent with the other one. The other with post-upgrade-image-tag didn't work with `ref`